framework/doc/content/syntax/FunctorMaterials/index.md

```diff
@@ -0,0 +1,108 @@
+# FunctorMaterials
+
+Functor material properties are properties that are evaluated on-the-fly.
+Along with multiple other classes of MOOSE objects, they are [Functors](syntax/Functors/index.md). This allows
+for the creation of objects with considerable inter-operability between systems.
+
+## Using functor materials
+
+Much like regular materials declare regular material properties, functor materials declare functor material properties.
+Following a producer/consumer model, `FunctorMaterials` produce functor material properties, which kernels, boundary conditions, other
+functor materials, etc, may consume.
+Both regular and functor material properties support a wide array of data types (scalar, vector, tensors,
+[automatic differentiation](automatic_differentiation/index.md) versions of all of these). Functor materials, because they are evaluated on-the-fly
+and call other functors on-the-fly as well, do not require dependency ordering.
+
+All functor materials support caching in some capacity. This can avoid expensive material property computations, but is
+disabled by default due to the potential memory cost. The reader is referred to the [Functors caching documentation](syntax/Functors/index.md#caching)
+
+!alert warning
+Functor materials are generally **NOT** compatible with regular materials, as regular material properties **CANNOT** be used
+in lieu of functor material properties. Functor material properties can only be used as regular material properties using a
+[MaterialFunctorConverter.md] to process them.
+
+Functor materials can be created within the `[FunctorMaterials]` block, and currently the `[Materials]` block as well,
+but that syntax will be deprecated.
+
+!alert note
+If a Functor is reported as missing by the simulation, and it was supposed to be created by a `FunctorMaterial`,
+you may use the [!param](/Debug/SetupDebugAction/show_functors) to get more information about what functors were
```

Comment:
```suggestion
you may use the [!param](/Debug/SetupDebugAction/show_functors) parameter to get more information about what functors were
```

Reply:
What I meant was that there's no indication that this is a parameter that needs to be in a `Debug` block.